jetty-security/src/test/java/org/eclipse/jetty/security/authentication/SpnegoAuthenticatorTest.java

```diff
@@ -62,6 +63,35 @@ public Server getServer()
                 return null;
             }
 
+            @Override
+            public boolean failed(Throwable x)
+            {
+                return false;
+            }
+
+            @Override
+            protected boolean eof()
+            {
+                return false;
+            }
+
+            @Override
+            public boolean needContent()
+            {
+                return false;
+            }
+
+            @Override
+            public HttpInput.Content produceContent()
+            {
+                return null;
+            }
+
+            @Override
+            public void failContent(Throwable failure)
+            {
+            }
+
```

Comment:
We could, but since every non-test channel must implement them all, I'm not certain it's wise. Maybe we could move them to a new interface and have `HttpChannel` only have to implement a single getter for the impl of that new itnerface?